test-suite/js/src/test/scala/org/scalajs/testsuite/library/LinkTimeIfTest.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.library
+
+import scala.scalajs.js
+import scala.scalajs.LinkingInfo._
+
+import org.junit.Test
+import org.junit.Assert._
+import org.junit.Assume._
+
+import org.scalajs.testsuite.utils.Platform
+
+class LinkTimeIfTest {
```

Comment:
It feels to me that a unit test of the evaluation method is much better suited to test all the individual evaluation (faster, less interference of other things).